src/debezium-service/templates/deployment.yaml

```diff
@@ -0,0 +1,167 @@
+                    for ((n=0;n<10;n++)); do
+                      jcmd $PID Thread.print -e > $loc/thread-dump-attempt-$n.txt;
+                      if [ $(wc -l < $loc/thread-dump-attempt-$n.txt) -gt 1 ]; then break; fi;
+                    done
+                    
+                    #10 retries to take heap histogram. Unsuccessful attempt has just 1 line with java process id in the output.
+                    for ((n=0;n<10;n++)); do
+                      jcmd $PID GC.class_histogram -all > $loc/heap-histogram-attempt-$n.txt;
+                      if [ $(wc -l < $loc/heap-histogram-attempt-$n.txt) -gt 1 ]; then break; fi;
+                    done
+                    
+                    jcmd $PID VM.native_memory  > $loc/native-memory-dump.txt; 
+                    
+                    #Dump latest chunk of JFR recording
+                    jcmd $PID JFR.dump name=jfrRecording filename=/opt/harness/dumps/${SERVICE_NAME}/${ENV_TYPE}/jfr_dumps/${POD_NAME}/container_termination_$(date +%Y_%m_%d_%H_%M_%S).jfr > $loc/jfr_done.txt
+                    
+                    jcmd $PID GC.heap_dump > $loc/heapdump.hprof;
+                    echo $(date '+%s') > $loc/end
+                    kill -15 $PID;
+          ports:
+            - name: debezium-http
+              containerPort: {{.Values.httpServerPort}}
+              protocol: "TCP"
+            - name: metrics
+              containerPort: {{.Values.PROMETHEUS_COLLECTOR_PORT}}
+              protocol: "TCP"
+          resources:
+            limits:
+              memory: {{.Values.podMemory}}
+            requests:
+              memory: {{.Values.podMemory}}
+              cpu: {{.Values.cpu | quote}}
+          envFrom:
+            - configMapRef:
+                name: {{.Values.name}}
+            - secretRef:
+                name: {{.Values.name}}
+          readinessProbe:
+            httpGet:
+              path: /api/health
+              port: {{.Values.httpServerPort}}
+            initialDelaySeconds: 60
+            timeoutSeconds: 10
+            periodSeconds: 10
+            failureThreshold: 20
+          livenessProbe:
+            httpGet:
+              path: /api/health
+              port: {{.Values.httpServerPort}}
+            initialDelaySeconds: 60
+            timeoutSeconds: 10
+            periodSeconds: 10
+            failureThreshold: 20
+          env:
+            - name: POD_NAME
+              valueFrom:
+                fieldRef:
+                  fieldPath: metadata.name
+            - name: SERVICE_NAME
+              value: debezium-service
+            - name: ENV_TYPE
+              value: {{.Values.envType}}
+            - name: TAG
+              value: {{.Values.codeVersion}}
+          volumeMounts:
+            - name: service-account
+              mountPath: /opt/harness/svc
+            - name: dumps
+              mountPath: /opt/harness/dumps
+      tolerations:
```

Comment:
check other charts for affinity, nodeSelector and tolerations, this is hardcoded and cannot be used